pkg/cmd/roachtest/tests/util.go

```diff
@@ -219,7 +220,14 @@ func setAdmissionControl(ctx context.Context, t test.Test, c cluster.Cluster, en
 // test uploads cockroach using `t.Cockroach` (instead of calling
 // t.StandardCockroach or t.RuntimeAssertionsCockroach directly).
 func UsingRuntimeAssertions(t test.Test) bool {
-	return t.Cockroach() == t.RuntimeAssertionsCockroach()
+	switch t.Spec().(*registry.TestSpec).CockroachBinary {
+	case registry.StandardCockroach:
+		return false
+	case registry.RuntimeAssertionsCockroach:
+		return true
+	default:
+		return t.Cockroach() == t.RuntimeAssertionsCockroach()
```

Comment:
Nit: maybe we could simplify/consolidate the above cases under `t.Cockroach`? `PutCockroach` does the same case analysis. Instead, `t.Cockroach` could just return the correct binary based on the specification.

Reply:
Done